external-crates/move/crates/move-analyzer/src/symbols.rs

```diff
@@ -1155,24 +1155,15 @@ pub fn get_symbols(
         None
     };
 
-    // get source files to be able to correlate positions (in terms of byte offsets) with actual
-    // file locations (in terms of line/column numbers)
+    let mut mapped_files: MappedFiles = MappedFiles::empty();
+
+    // Hash dependencies so we can check if something has changed.
     let source_files = file_sources(&resolution_graph, overlay_fs_root.clone());
-    let mut files = SimpleFiles::new();
-    let mut file_id_mapping = HashMap::new();
-    let mut file_id_to_lines = HashMap::new();
-    let mut file_name_mapping = BTreeMap::new();
     let mut hasher = Sha256::new();
-    for (fhash, (fname, source, is_dep)) in &source_files {
-        if *is_dep {
-            hasher.update(fhash.0);
-        }
-        let id = files.add(*fname, source.clone());
-        file_id_mapping.insert(*fhash, id);
-        file_name_mapping.insert(*fhash, PathBuf::from(fname.as_str()));
-        let lines: Vec<String> = source.lines().map(String::from).collect();
-        file_id_to_lines.insert(id, lines);
-    }
+    source_files
```

Comment:
Now that with this PR we will be using mapped files from the compiler it's a shame that we read and hash their content here as well. Would it make sense to (and be less expensive?) to do a different validation here (e.g., read and hash last-modified date for each file and compare these rather then the content)?

Reply:
Let's do that in a different PR. I have some ideas for making this work.